understanding/sharding.md

```diff
@@ -2,29 +2,210 @@
-* In order to do anything across shards you need to worry about using `broadcastEval` and such \(tutorial coming soon!\).
-* Sharded bots often gain very marginal performance increase and might even use _more_ memory due to using more node processes.
+## Internal Sharding
+
+`internal` sharding is the method by which a bot's code creates multiple shard connections to the Discord API *within a single process*. This means that all the guilds, channels, and users on one shard will be available to another shard via a direct call (e.g. `client.guilds.cache.get('GUILD_ID')`). Due to the large memory size the single bot process will grow to using this style of sharding, it is not ideal for bots with many guilds.
+
+### Internal Sharding Caveats
+
+* Internally sharded bots often gain very marginal performance increase and will use *more memory*.
 * If you're using any sort of database or connection, multiple shards may cause issues with multiple processes connecting to a single end point.
 
-## Example Sharding Code
+### Internally Sharded Client
+
+If you would like to use this, adjust the Client options in your main bot file where you define your client like so:
+
+```javascript
+/*
+    The following code goes into your main bot file.
+*/
+
+// Include discord.js ShardingManger
+const Discord = require("discord.js");
+
+// When we define our client, we include the property "shardCount"
+// and set it to 'auto' to allow the client to automatically create
+// the correct number of shards.
+// If you would like to have a different number of shards, you may
+// also set this to a number.
+const client = new Discord.client({ shardCount: 'auto'});
+```
+
+## Traditional Sharding
+
+`traditional` sharding is the method by which a bot's code spawns individual child processes via a main shard manager process, each child process being one shard of the bot. When using this style of sharding, guilds, channels, and users on one shard will *not* be available to another via direct call (e.g. `client.guilds.cache.get('GUILD_ID')`) because each shard is in a separate process.
+
+This style of sharding is ideal for larger bots, or bots that need to be scalable to allow for future growth. The rest of this page will discuss [how to make use of traditional sharding](#example-sharding-manager-code) and [how to share information between shards](#sharding-information-between-shards).
+
+To learn how to make use of this, read on!
 
+### Traditional Sharding Caveats
+
+* Collections do not cache data from all shards, so you can't grab data from a guild in another shard easily.
+* In order to do anything across shards you need to worry about using [`fetchClientValues`](#fetchclientvalues) and [`broadcastEval`](#broadcasteval) \(Examples and explanation below\).
+And again:
+* Traditionally sharded bots often gain very marginal performance increase and might even use *more memory* due to using more node processes.
```

Comment:
So both internal sharding and traditional sharding use more memory, so the question is, what are the *advantages* of using internal, or traditional, sharding? Caveats are one thing, but why would one do it other than to just satisfy Discord's requirements?

Reply:
Internal sharding comes with near to no memory overhead, whereas separate processes still have to load all the dependencies to memory, which in large bots can be 20-50mb.

Therefore, this is a valid caveat of traditional sharding, specially when they stack up.